metals/src/main/scala/scala/meta/internal/implementation/ImplementationProvider.scala

```diff
@@ -177,29 +180,37 @@ final class ImplementationProvider(
     }
 
     import TokenEditDistance.fromBuffer
-
+    val allLocations = new ConcurrentLinkedQueue[Location]
     for {
       classContext <- inheritanceContext.toIterable
       plainParentSymbol <- classContext.findSymbol(symbol).toIterable
       parentSymbol = addParameterSignatures(plainParentSymbol, classContext)
       symbolClass <- classFromSymbol(parentSymbol, classContext.findSymbol)
-      (file, locations) <- findImplementation(symbolClass.symbol, classContext)
-      implPath = AbsolutePath(file)
-      implDocument <- findSemanticdb(implPath).toIterable
-      distance = fromBuffer(implPath, implDocument.text, buffer)
-      implLocation <- locations
-      implReal = implLocation.toRealNames(symbolClass, translateKey = true)
-      implSymbol <- findImplementationSymbol(
-        parentSymbol,
-        implDocument,
-        symbolClass,
-        classContext,
-        implReal
-      )
-      implOccurrence <- findDefOccurrence(implDocument, implSymbol, source)
-      range <- implOccurrence.range
-      revised <- distance.toRevised(range.toLSP)
-    } yield new Location(file.toUri.toString, revised)
+      parallelMap = findImplementation(
+        symbolClass.symbol,
+        classContext
+      ).par
```

Comment:
I'm not sure how the parallel map works, I generally only use parallel arrays

```scala
val map = findImplentation(...)
keys.toArray.par.foreach { file =>
  val locations = map(file)
  // ...
}
```

Reply:
Shouldn't it work basically the same?

Reply:
From what I checked the only real issue is with collections like List, which are sequential by nature. Map should be fine since it allows easily for concurrent access.

Reply:
After a bit more research, Arrays seem to be most efficient really, so changed to that.